src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java

```diff
@@ -109,16 +108,23 @@ private Future<List<Piece>> createPieces(OpenOrderPieceHolder holder, boolean is
   public Future<Piece> createPiece(Piece piece, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
     logger.debug("createPiece start");
     return purchaseOrderStorageService.getCompositeOrderByPoLineId(piece.getPoLineId(), requestContext)
-      .compose(order -> protectionService.isOperationRestricted(order.getAcqUnitIds(), ProtectedOperationType.CREATE, requestContext)
-        .map(v -> order))
+      .compose(vVoid ->
+        titlesService.getTitleById(piece.getTitleId(), requestContext)
+          .compose(title ->
+            protectionService.isOperationRestricted(title.getAcqUnitIds(), ProtectedOperationType.CREATE, requestContext)
+              .map(v -> vVoid)))
       .compose(order -> openOrderUpdateInventory(order.getCompositePoLines().get(0), piece, isInstanceMatchingDisabled, requestContext))
       .compose(v -> pieceStorageService.insertPiece(piece, requestContext));
   }
 
   public Future<Void> updatePieceRecord(Piece piece, RequestContext requestContext) {
     Promise<Void> promise = Promise.promise();
     purchaseOrderStorageService.getCompositeOrderByPoLineId(piece.getPoLineId(), requestContext)
-      .compose(order -> protectionService.isOperationRestricted(order.getAcqUnitIds(), ProtectedOperationType.UPDATE, requestContext))
+      .compose(vVoid ->
+        titlesService.getTitleById(piece.getTitleId(), requestContext)
+          .compose(title ->
+            protectionService.isOperationRestricted(title.getAcqUnitIds(), ProtectedOperationType.UPDATE, requestContext)
+              .map(v -> vVoid)))
```

Comment:
so I see that we don't use order in the next operation. Can we remove this line?

Reply:
Deleted